### ISSUE TITLE
[feature] CMakeToolchain: prepend CMAKE_CURRENT_LIST_DIR instead of CMAKE_BINARY_DIR

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#8808]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

Currently, toolchain file generated by CMakeToolchain sets `CMAKE_MODULE_PATH` and `CMAKE_PREFIX_PATH` like this:

```cmake
    # To support the cmake_find_package generators
    set(CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR} ${CMAKE_MODULE_PATH})
    set(CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR} ${CMAKE_PREFIX_PATH})
```

It would be more robust to prepend `CMAKE_CURRENT_LIST_DIR` since `conan install` could be called from an other directory than root build tree.
Usually I call `conan install` in a subfolder (I like to call it `conan_generators`) to isolate specific conan generated build files from standard build artifacts.
I can live with this behavior, I inject absolute path of this subfolder (or any folder on my system where Ive called `conan install`) into `CMAKE_MODULE_PATH` & `CMAKE_PREFIX_PATH` through  cmake command line, but it would be nice if toolchain file was more robust.

**Comment by memsharded** [comment]
But using CMAKE_CURRENT_LIST_DIR will have the same issues, or even worse, because it will reference the CMakeLists folder, which is usually in the source space, and it means that it should expect to have all generated xxx-config.cmake files in the source folder, not the temporary build one. What am I missing here?

In any case, the solution to this is more a explicit modeling and parameterization of the folder where generators will put the files. This is the only way to achieve reproducibility and robustness, irrespective of local flow or cache, and cwd issues. We have already started to work on this.

**Comment by SpaceIm** [comment]
AFAIK, `CMAKE_CURRENT_LIST_DIR` doesn't refer to any `CMakeLists.txt` folder, but to the local folder where the cmake file referencing this variable lives. Since toolchain files is generated in the same folder than others conan generated file, it's exactly the folder you want to append in CMAKE_MODULE_PATH & CMAKE_PREFIX_PATH.

**Comment by memsharded** [comment]
Oh, yes, you are right, I misunderstood. Sent PR #8810

**Comment by SpaceIm** [comment]
No problem, there was a typo in the code snippet of the first message referring to current behavior, maybe it was not very clear.

**Comment by memsharded** [comment]
Merged https://github.com/conan-io/conan/pull/8810, will be released next 1.36
